llumnix/backends/bladellm/llm_engine.py

```diff
@@ -91,20 +99,33 @@ def __init__(self,
                  request_output_queue_type: QueueType,
                  resp_queue: asyncio.Queue,
                  metrics_queue: asyncio.Queue,
-                 backend_type: BackendType) -> None:
-        scheduling_strategy = PlacementGroupSchedulingStrategy(
-            placement_group=placement_group,
-            placement_group_bundle_index=0,
-            placement_group_capture_child_tasks=True,
-        )
+                 backend_type: BackendType,
+                 request_output_forwarding_mode: RequestOutputForwardingMode,
+                 drop_request_coro: Coroutine) -> None:
         self.instance_id = instance_id
-        self.async_put_queue_actor = ray.remote(
-            num_cpus=1,
-            num_gpus=NUM_GPUS_BLADELLM_GPU_ACTOR,
-            scheduling_strategy=scheduling_strategy,
-            name=f"AsyncPutQueueActor_{instance_id}"
-        )(AsyncPutQueueActor).remote(instance_id, request_output_queue_type, backend_type)
-        self.put_queue_args_queue: asyncio.Queue = resp_queue
+        self.request_output_queue_type = request_output_queue_type
+        self.request_output_forwarding_mode = request_output_forwarding_mode
+        if self.request_output_forwarding_mode == RequestOutputForwardingMode.ACTOR:
+            scheduling_strategy = PlacementGroupSchedulingStrategy(
+                placement_group=placement_group,
+                placement_group_bundle_index=0,
+                placement_group_capture_child_tasks=True,
+            )
+            self.async_put_queue_actor = ray.remote(
+                num_cpus=1,
+                num_gpus=NUM_GPUS_BLADELLM_GPU_ACTOR,
+                scheduling_strategy=scheduling_strategy,
+                name=f"AsyncPutQueueActor_{instance_id}"
+            )(AsyncPutQueueActor).remote(instance_id, request_output_queue_type, backend_type)
+        else:
+            self.put_queue_args_queue = queue.Queue()
+            self.main_loop = asyncio.get_event_loop()
+            self.drop_request_coro = drop_request_coro
```

Comment:
why use "drop" here and "abort" elsewhere

Reply:
in bladellm drop, in vllm abort, and here just follow the original name of engine function.